workspace/exercise.go

```diff
@@ -59,3 +65,46 @@ func (e Exercise) HasMetadata() (bool, error) {
 	}
 	return false, err
 }
+
+// HasLegacyMetadata checks for the presence of a legacy exercise metadata file.
+// If there is no such file, it could also be an unrelated directory.
+func (e Exercise) HasLegacyMetadata() (bool, error) {
+	_, err := os.Lstat(e.LegacyMetadataFilepath())
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	if err == nil {
+		return true, nil
+	}
+	return false, err
+}
+
+// MigrateLegacyMetadataFile migrates a legacy metadata to the modern location.
+// This is a noop if the metadata file isn't legacy.
+// If both legacy and modern metadata files exist, the legacy file will be deleted.
+func (e Exercise) MigrateLegacyMetadataFile() (string, error) {
+	var str string
+	legacyMetadataFilepath := e.LegacyMetadataFilepath()
+	metadataFilepath := e.MetadataFilepath()
+
+	if _, err := os.Lstat(legacyMetadataFilepath); err != nil {
+		return "", nil
```

Comment:
Is the assumption here that if we get an error, it's an `os.IsNotExist` error? Is that a safe assumption?

Reply:
Yeah, this is problematic. I replaced this by calling exercise.HasMetadata() since it contains the logic needed here. The trade-off is that this adds excess calls to MetadataFilepath() but it might be worth it for improved readability and avoiding duplication?

Reply:
> The trade-off is that this adds excess calls to MetadataFilepath() but it might be worth it for improved readability and avoiding duplication?

Yeah, I wouldn't worry about that. If we find that we have performance problems we can do proper profiling etc to see what's up and fix it 👍